src/hdmf/build/map.py

```diff
@@ -431,6 +431,8 @@ def convert_dtype(cls, spec, value):
                 ret = np.asarray(value).astype(dtype_func)
                 ret_dtype = ret.dtype.type
         elif isinstance(value, (tuple, list)):
+            if len(value) == 0:
+                return value, spec['dtype']
```

Comment:
Why is this not `return value, cls.__spec[spec_dtype]` ? This method is meant to return the `numpy.dtype` object. The way this line is written will return a `str`

Reply:
that makes a good amount of sense. Will fix

Reply:
hmm, ObjectMapper seems not to have a `__spec` attribute. It looks like `__resolve_dtype` works on an object and a type spec to produce a numpy dtype. Since this is handling the degenerate case where we only have a spec, does it work for this line to just read:
```Python
return value, np.dtype(spec_dtype)
```
?

Reply:
@ajtritt one other thing - when testing pynwb against latest hdmf (with these changes or not) I notice that there are about 5 test failures. Are these known & tracked? What is the method for testing pynwb against the latest hdmf?

Reply:
My bad, I meant.

```python
return value, spec_dtype
```
the `str` -> `np.dtype` gets handled on 421, no need to do it yourself.

Reply:
I have not experience the errors in NWB with the latest HDMF, but lets take that conversation elsewhere